.github/workflows/release.yml

```diff
@@ -0,0 +1,67 @@
+name: Build and upload to PyPI
+
+# Build on every branch push, tag push, and pull request change:
+on: [push, pull_request]
+
+jobs:
+  build_wheels:
+    name: Build wheels on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
```

Comment:
Something we use on https://github.com/astropy/pytest-doctestplus/blob/main/.github/workflows/publish.yml to avoid building wheels for every commit/PR. This leaves the possibility to build wheels for a PR by applying a "Build wheels" label.
```suggestion
    runs-on: ${{ matrix.os }}
    if: ((github.event_name == 'push' && startsWith(github.ref, 'refs/tags')) || contains(github.event.pull_request.labels.*.name, 'Build wheels'))
```

Reply:
I'm kind of wondering don't we want to build wheels for every pr? Isn't that basically part of the checks we want to run whenever we are thinking about merging something?

Reply:
Since there are not so many PRs on this repo, that's indeed something we could do.